src/Analyzer.JsonRuleEngine/Schema/AllOfExpressionDefinition.cs

```diff
@@ -24,9 +24,20 @@ internal class AllOfExpressionDefinition : ExpressionDefinition
         /// </summary>
         /// <param name="jsonLineNumberResolver">An <see cref="ILineNumberResolver"/> to
         /// pass to the created <see cref="Expression"/>.</param>
+        /// <param name="isNegative">Whether to negate the result of the evaluation.</param>
         /// <returns>The AllOfExpression.</returns>
-        public override Expression ToExpression(ILineNumberResolver jsonLineNumberResolver)
-            => new AllOfExpression(this.AllOf.Select(e => e.ToExpression(jsonLineNumberResolver)).ToArray(), GetCommonProperties(jsonLineNumberResolver));
+        public override Expression ToExpression(ILineNumberResolver jsonLineNumberResolver, bool isNegative = false)
+        {
+            if (!isNegative)
+            {
+                return new AllOfExpression(this.AllOf.Select(e => e.ToExpression(jsonLineNumberResolver, isNegative)).ToArray(), GetCommonProperties(jsonLineNumberResolver));
+            }
+            else
+            {
+                // De Morgan's Law
+                return new AnyOfExpression(this.AllOf.Select(e => e.ToExpression(jsonLineNumberResolver, isNegative)).ToArray(), GetCommonProperties(jsonLineNumberResolver));
```

Comment:
For De Morgan's Law wouldn't the individual items then each be negated? !(A&B) === !A | !B

e.g. `return new AnyOfExpression(this.AllOf.Select(e =>e.ToExpression(jsonLineNumberResolver, !isNegative)...`

Tests will cover this so will be straightened out. :)

Reply:
It's what Johnathon said. If it helps we can hardcode the values, or we can just add a comment saying that isNegative is true in this case.